plugins/connection/aws_ssm.py

```diff
@@ -53,6 +53,11 @@
     vars:
     - name: ansible_aws_ssm_plugin
     default: '/usr/local/bin/session-manager-plugin'
+  profile:
+    description: Sets AWS profile to use.
+    vars:
+    - name: ansible_aws_ssm_profile
+    default: ''
```

Comment:
I think this must be set to `None`, even if it was defined as `''` in source code before.

Reply:
hmm no. I wonder if the section

* profile
* secret_access_key
* access_key_id
* session_token
can be dropped at all when the documentation gets

```
extends_documentation_fragment:
- amazon.aws.aws
```

appended.

Reply:
`description:` and `default:` should not be re-declared here, that extends/overwrites the values in https://github.com/ansible-collections/amazon.aws/blob/main/plugins/doc_fragments/aws.py#L64-L71. It should just be
```.yaml

  profile:
    vars:
    - name: ansible_aws_ssm_profile
```
Then the Ansible variable is added to the declaration from the document fragment.

Also, more precisely, the entry should look like:
```.yaml

  profile:
    vars:
    - name: ansible_aws_ssm_profile
      version_added: 1.3.0
```
(Assuming the next release this is in will be 1.3.0, since according to https://galaxy.ansible.com/community/aws the current release is 1.2.1.)

Reply:
Thank you for the review, I've amended this section based on the comments. Since there's already references to `1.3.0`, I've set `version_added` to that.

Reply:
Oh, but CI is now complaining about the lack of a description field... Is that ok then? Doesn't seem like something I should ignore but I'm unclear on the right approach here.

Reply:
Oh, sorry, I think I somehow confused things. I saw @markuman mention
```
extends_documentation_fragment:
- amazon.aws.aws
```
and apparently assumed that's already there. If it would be there, omitting the description is the best choice. Unfortunately that's not the case (and adding the doc fragment requires some more refactoring/cleaning up), so I guess the best is to re-add the `description:`. Just don't re-add the `default:`, it's better without :) (and of course keep the `version_added`)

Sorry!!!

Reply:
No worries, fixed it back up and returned a description :sweat_smile: 

Looks like CI is happy now!